src/main/javascript/router.js

```diff
@@ -154,5 +207,23 @@ window.SwaggerUiRouter = Backbone.Router.extend({
         var pathname = location.pathname.substring(0, location.pathname.lastIndexOf('/'));
 
         return host.protocol + '//' + host.host + pathname.replace('swagger', 'api/swagger/docs/v1') + '?_=' + Date.now();
+    },
+
+    getParameterByName: function(name, url) {
+        if (!url) {
+            url = window.location.href;
+        }
+        name = name.replace(/[\[\]]/g, '\\$&');
```

Comment:
Could you please add some comments to understand in future how to extend this function? Details about what regex describes in this example.